### ISSUE TITLE
Starlark impl function naming convention not clear

### ISSUE THREAD

**Issue by xingao267** [bazelbuild/bazel#7699]
It seems not consistent in a lot of places where `_impl` or `<rule_name>_impl` should be used as  convention for the implementation function name. According to doc [here](https://docs.bazel.build/versions/master/skylark/rules.html#implementation-function), it should be `<rule_name>_impl`, but the [example](https://github.com/bazelbuild/examples/blob/master/rules/attributes/printer.bzl) given in the same section uses `_impl`. And the whole [doc](https://docs.bazel.build/versions/master/skylark/rules.html) uses a mix of the two. It caused confusion here https://github.com/bazelbuild/rules_docker/pull/724#discussion_r264527597

**Comment by aiuto** [comment]
Yes. The example is wrong.  It should be <rule_name>_impl so that you can have multiple rules in a single .bzl file. Otherwise the naming would be different for single rule vs. multi-rule files.

We always appreciate PRs to correct clear mistakes. Otherwise we'll get around to this in a documentation cleaning pass.
